libs/labelbox/src/labelbox/schema/task.py

```diff
@@ -214,3 +227,170 @@ def get_task(client, task_id):
         task: Task = tasks[0]
         task._user = user
         return task
+
+
+class DataUpsertTask(Task):
+    __max_donwload_size: Final = MAX_DATAROW_PER_API_OPERATION
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self._user = None
+
+    @property
+    def result(self) -> Optional[List[Dict[str, Any]]]:  # type: ignore
+        if self.status == "FAILED":
+            raise ValueError(f"Job failed. Errors : {self.errors}")
+        return self._results_as_list()
+
+    @property
+    def errors(self) -> Optional[List[Dict[str, Any]]]:  # type: ignore
+        return self._errors_as_list()
+
+    @property
+    def created_data_rows(  # type: ignore
+            self) -> Optional[List[Dict[str, Any]]]:
+        return self.result
+
+    @property
+    def failed_data_rows(  # type: ignore
```

Comment:
had to do `# type: ingnore` because linter complained due to a change in return type (see the above comment re: need to refactor Task)